execute/plugin.go

```diff
@@ -272,7 +276,13 @@ func (p *Plugin) Observation(
 			return types.Observation{}, fmt.Errorf("unable to process token data %w", err1)
 		}
 
-		return exectypes.NewObservation(groupedCommits, messages, tkData, nil, discoveryObs).Encode()
+		costlyMessages, err := p.costlyMessageObserver.Observe(ctx, messages)
```

Comment:
Trying to understand this: at this point, we have messages from potentially many commit reports, and we filter out those that are "too costly" to execute, even after boosting. Next round comes around, say a commit report has some subset of its messages executed, the remaining not executed due to being too costly. We pick up from there again, and we determine if they're still too costly, etc?

Reply:
Yes, that's correct